schema_parse.go

```diff
@@ -199,20 +208,21 @@ func parsePrimitiveLogicalType(typ Type, lt string, prec, scale int) LogicalSche
 	}
 
 	if typ == Bytes && ltyp == Decimal {
-		return parseDecimalLogicalType(-1, prec, scale)
+		return parseDecimalLogicalType(-1, otherProps)
 	}
 
-	return nil
+	return nil // otherwise, not a recognized logical type
 }
 
 type recordSchema struct {
-	Type      string           `mapstructure:"type"`
-	Name      string           `mapstructure:"name"`
-	Namespace string           `mapstructure:"namespace"`
-	Aliases   []string         `mapstructure:"aliases"`
-	Doc       string           `mapstructure:"doc"`
-	Fields    []map[string]any `mapstructure:"fields"`
-	Props     map[string]any   `mapstructure:",remain"`
+	Type        string           `mapstructure:"type"`
+	LogicalType string           `mapstructure:"logicalType"`
```

Comment:
If this didnt exist, `logicalType` would automatically go into props, and the call to `preserveLogicalType` would not be needed, no?

Reply:
That's true. It's presence was solely for type-checking: parsing the map into this struct will require that the `logicalType` attribute, if present, be a string. I could remove this line from here (and the other places that don't actually do anything with the logical type other than add to properties) but then I think we'd need to replace the call to `preserveLogicalType` with something like `checkLogicalTypeIsString(props)`.

Reply:
Removed these lines from various structs (record, enum, array, and map) and replaced with more explicit type check.